sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -356,6 +356,8 @@ case class Not(child: Expression)
 case class InSubquery(values: Seq[Expression], query: ListQuery)
   extends Predicate with Unevaluable {
 
+  assert(query.plan.output.nonEmpty, "The subquery must have output.")
```

Comment:
I mean adding an assert in `RowLevelOperationRuntimeGroupFiltering`, where we create `InSubquery`...

Reply:
The size of `buildKeys` is always equal to size of `scan.filterAttributes`, is assert required?